File: examples/javascript/upify_main.js
const express = require('express');
const { getWeatherData } = require('./index');

const app = express();

app.get('/', async (req, res) => {
    if (!req.query.city) {
        console.log("Specify a city in the query string");
        return res.status(400).json({ error: "Specify a city in the query string" });
    }

    const city = req.query.city;
    console.log("Got a weather request for " + city);
    
    try {
        const responseData = await getWeatherData(city);
        res.status(200).json(responseData);
    } catch (error) {
        console.error("Error getting weather data:", error);
        res.status(500).json({ error: error.message });
    }
});

module.exports = app;

File: examples/python/upify_main.py
from flask import Flask, request, jsonify
from main import get_weather_data

app = Flask(__name__, instance_path='/tmp')

@app.route('/')
def handler():
    city = request.args.get("city")
    if city is None:
        return jsonify({"error": "Specify a city in the query string"}), 400

    try:
        response_data = get_weather_data(city)
        return jsonify(response_data), 200
    except Exception as e:
        return jsonify({"error": f"Failed to get weather data: {repr(e)}"}), 500